niworkflows/utils/bids.py

```diff
@@ -245,3 +231,71 @@ def get_metadata_for_nifti(in_file):
                 json.loads(json_file_path.read_text()))
 
     return merged_param_dict
+
+
+def group_multiecho(bold_sess):
+    """
+    Multiplexes multi-echo EPIs into arrays. Dual-echo is a special
+    case of multi-echo, which is treated as single-echo data.
+
+    >>> bold_sess = ["sub-01_task-rest_echo-1_run-01_bold.nii.gz",
+    ...              "sub-01_task-rest_echo-2_run-01_bold.nii.gz",
+    ...              "sub-01_task-rest_echo-1_run-02_bold.nii.gz",
+    ...              "sub-01_task-rest_echo-2_run-02_bold.nii.gz",
+    ...              "sub-01_task-rest_echo-3_run-02_bold.nii.gz",
+    ...              "sub-01_task-rest_run-03_bold.nii.gz"]
+    >>> group_multiecho(bold_sess)
+    ['sub-01_task-rest_echo-1_run-01_bold.nii.gz',
+     'sub-01_task-rest_echo-2_run-01_bold.nii.gz',
+    ['sub-01_task-rest_echo-1_run-02_bold.nii.gz',
+     'sub-01_task-rest_echo-2_run-02_bold.nii.gz',
+     'sub-01_task-rest_echo-3_run-02_bold.nii.gz'],
+     'sub-01_task-rest_run-03_bold.nii.gz']
+
+    >>> bold_sess.insert(2, "sub-01_task-rest_echo-3_run-01_bold.nii.gz")
+    >>> group_multiecho(bold_sess)
+    [['sub-01_task-rest_echo-1_run-01_bold.nii.gz',
+      'sub-01_task-rest_echo-2_run-01_bold.nii.gz',
+      'sub-01_task-rest_echo-3_run-01_bold.nii.gz'],
+     ['sub-01_task-rest_echo-1_run-02_bold.nii.gz',
+      'sub-01_task-rest_echo-2_run-02_bold.nii.gz',
+      'sub-01_task-rest_echo-3_run-02_bold.nii.gz'],
+      'sub-01_task-rest_run-03_bold.nii.gz']
+
+    >>> bold_sess += ["sub-01_task-beh_echo-1_run-01_bold.nii.gz",
+    ...               "sub-01_task-beh_echo-2_run-01_bold.nii.gz",
+    ...               "sub-01_task-beh_echo-1_run-02_bold.nii.gz",
+    ...               "sub-01_task-beh_echo-2_run-02_bold.nii.gz",
+    ...               "sub-01_task-beh_echo-3_run-02_bold.nii.gz",
+    ...               "sub-01_task-beh_run-03_bold.nii.gz"]
+    >>> group_multiecho(bold_sess)
+    [['sub-01_task-rest_echo-1_run-01_bold.nii.gz',
+      'sub-01_task-rest_echo-2_run-01_bold.nii.gz',
+      'sub-01_task-rest_echo-3_run-01_bold.nii.gz'],
+     ['sub-01_task-rest_echo-1_run-02_bold.nii.gz',
+      'sub-01_task-rest_echo-2_run-02_bold.nii.gz',
+      'sub-01_task-rest_echo-3_run-02_bold.nii.gz'],
+      'sub-01_task-rest_run-03_bold.nii.gz',
+      'sub-01_task-beh_echo-1_run-01_bold.nii.gz',
+      'sub-01_task-beh_echo-2_run-01_bold.nii.gz',
+     ['sub-01_task-beh_echo-1_run-02_bold.nii.gz',
+      'sub-01_task-beh_echo-2_run-02_bold.nii.gz',
+      'sub-01_task-beh_echo-3_run-02_bold.nii.gz'],
+      'sub-01_task-beh_run-03_bold.nii.gz']
+
+    """
```

Comment:
I don't think we run doctests.

Reply:
I think I added them when I brought the code from fmriprep:

https://github.com/poldracklab/niworkflows/blob/master/.circleci/config.yml#L69-L126

Reply:
Oh, you're right. It's in:

https://github.com/poldracklab/niworkflows/blob/f69a835c33f2c465cda57bd8533e2d4030657ba3/tox.ini#L12

Reply:
Would be easier to be check with coverage turned on. :-)

Reply:
Definitely - #273